### PR TITLE
Add french locale

### LINKED ISSUE #280: French language support
🇫🇷

### PR DESCRIPTION
**Pull request by izeau**
Happy to help 😄 

Question! There’s no AM/PM indications in french, except for the long versions (“du matin”, “de l’après-midi”) which are more or less used depending on the region.. Should I include those?

Closes #280.

### CODE REVIEW

**Comment by LeshaKoss** [comment]
@izeau thank you!

> Question! There’s no AM/PM indications in french, except for the long versions (“du matin”, “de l’après-midi”) which are more or less used depending on the region.. Should I include those?

Probably. Think of a, A and aa formats like of indicators of the time of day when used with 12 hour clock, for example in Russian locale these formats can output 4 results ("in the night", "in the morning", "in the afternoon", "in the evening")
https://github.com/date-fns/date-fns/blob/master/src/locale/ru/build_format_locale/index.js#L40

**Comment by LeshaKoss** [comment]
@izeau on the other hand, moment.js uses just am/pm and AM/PM and maybe that's what we should do too 😅

**Comment by LeshaKoss** [comment]
@izeau 

>  (“du matin”, “de l’après-midi”) which are more or less used depending on the region

do frankophones use these commonly? I so, let's leave `meridiemUppercase`/`A` and `meridiemLowercase`/`a` as they are and use full versions for `meridiemFull`/`aa`. What do you think?

File: docs/i18n.md [hunk 1]
```diff
@@ -88,6 +88,7 @@ date-fns currently supports:
    (`ja`; kudos to Thomas [@DeMuu](https://github.com/DeMuu) Eilmsteiner)
 7. [Spanish](https://github.com/date-fns/date-fns/tree/master/src/locale/es)
    (`es`; kudos to Juan [@juanangosto](https://github.com/juanangosto) Angosto)
+8. [French](https://github.com/date-fns/date-fns/tree/master/src/locale/fr)
```
Comment (LeshaKoss) [inline docs/i18n.md]: This line causes merge conflict. Please remove it, I'll add it myself after the merge

**Comment by izeau** [comment]
French speakers do use these but mostly when speaking casually and where the exact time cannot be determined by context. When written times are almost always HH:mm.

File: src/locale/fr/build_format_locale/index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  // Generally, formatted dates should look like they are in the middle of a sentence.
+  // e.g. in Spanish language the weekdays and months should be in the lowercase.
+  var months3char = ['janv.', 'févr.', 'mars', 'avr.', 'mai', 'juin', 'juill.', 'août', 'sept.', 'oct.', 'nov.', 'déc.']
+  var monthsFull = ['janvier', 'février', 'mars', 'avril', 'mai', 'juin', 'juillet', 'août', 'septembre', 'octobre', 'novembre', 'décembre']
+  var weekdays2char = ['di', 'lu', 'ma', 'me', 'je', 've', 'sa']
+  var weekdays3char = ['dim.', 'lun.', 'mar.', 'mer.', 'jeu.', 'ven.', 'sam.']
+  var weekdaysFull = ['dimanche', 'lundi', 'mardi', 'mercredi', 'jeudi', 'vendredi', 'samedi']
+  var meridiemUppercase = ['AM', 'PM']
+  var meridiemLowercase = ['am', 'pm']
+  var meridiemFull = ['a.m.', 'p.m.']
```
Comment (LeshaKoss) [inline src/locale/fr/build_format_locale/index.js]: @izeau hello again! Let's make it `var meridiemFull = ['du matin', 'de l’après-midi']`

`a`/`A`/`aa` aren't strictly AM and PM but a way to specify which part of day it is, when used with 12 hour clock

For example, in Russian they do this:
`format(date, 'h a')` => `'7 вечера'` (like "7 in the afternoon")

In moment.js' French locale `a` and `A` are `am`/`pm` and `AM`/`PM` so let's leave `meridiemUppercase` and `meridiemLowercase` as they are, but change `meridiemFull` to what you suggested.

(please do this 🙏)

**Comment by izeau** [comment]
@LeshaKoss @kossnocorp I’ve just updated the PR, can you please take a look?

**Comment by LeshaKoss** [comment]
@izeau thanks a lot for your contribution! 🇫🇷 French locale was released with [1.17.0](https://github.com/date-fns/date-fns/releases/tag/v1.17.0) (diff: https://github.com/date-fns/date-fns/compare/v1.16.0...v1.17.0)

**Pull request merged**